CMakeLists.txt

```diff
@@ -150,25 +147,86 @@ if(NOT Qt6_FOUND)
     find_package(Qt5 5.15 COMPONENTS Core Gui Test REQUIRED)
 endif()
 get_target_property(QMAKE Qt::qmake IMPORTED_LOCATION)
-set(CARGO_ENV "QMAKE=${QMAKE}")
+set(CARGO_ENV "QMAKE=set:${QMAKE}")
+set(RUNTIME_ENV "")
+
+# On windows, Qt dll needs to be in the PATH for the tests to run
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    execute_process(
+        COMMAND ${QMAKE} -query QT_INSTALL_BINS
+        OUTPUT_VARIABLE QT_INSTALL_BINS
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    execute_process(
+        COMMAND ${QMAKE} -query QT_INSTALL_PLUGINS
+        OUTPUT_VARIABLE QT_INSTALL_PLUGINS
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    execute_process(
+        COMMAND ${QMAKE} -query QT_INSTALL_QML
+        OUTPUT_VARIABLE QT_INSTALL_QML
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    list(
+        APPEND
+        RUNTIME_ENV
+        "PATH=path_list_append:${QT_INSTALL_BINS}"
+        "QT_PLUGIN_PATH=path_list_append:${QT_INSTALL_PLUGINS}"
+        "QML_IMPORT_PATH=path_list_append:${QT_INSTALL_QML}"
+        "QML2_IMPORT_PATH=path_list_append:${QT_INSTALL_QML}"
+    )
+    list(APPEND CARGO_ENV ${RUNTIME_ENV})
+endif()
 
 # Create helper method which adds relevent cargo tests for a given manifest
 function(add_test_cargo TEST_NAME_PREFIX MANIFEST_PATH ADD_DOCTESTS)
     # Add cargo as a test
-    add_test(NAME ${TEST_NAME_PREFIX}_cargo_tests COMMAND cargo test --all-features --all-targets --manifest-path ${MANIFEST_PATH})
-    set_property(TEST ${TEST_NAME_PREFIX}_cargo_tests PROPERTY ENVIRONMENT ${CARGO_ENV})
+    set(CARGO_TEST_NAME ${TEST_NAME_PREFIX}_cargo_tests)
+    add_test(NAME ${CARGO_TEST_NAME}
+             COMMAND cargo test --all-features --all-targets --manifest-path
+                     ${MANIFEST_PATH}
+    )
+    set_tests_properties(
+        ${CARGO_TEST_NAME} PROPERTIES ENVIRONMENT_MODIFICATION "${CARGO_ENV}"
```

Comment:
Because on windows PATH is separated by `;` not `:`.
Cmake strings are separated by `;`. So having a list of environment variable containing `;` is not possible